hil-test/tests/spi_half_duplex_write_psram.rs

```diff
@@ -85,13 +85,20 @@ mod tests {
     #[timeout(3)]
     fn test_spi_writes_are_correctly_by_pcnt(ctx: Context) {
         const DMA_BUFFER_SIZE: usize = 4;
-        const DMA_ALIGNMENT: DmaBufBlkSize = DmaBufBlkSize::Size32;
+        const DMA_ALIGNMENT: ExternalBurstConfig = ExternalBurstConfig::Size32;
         const DMA_CHUNK_SIZE: usize = 4096 - DMA_ALIGNMENT as usize;
 
         let (_, descriptors) = dma_descriptors_chunk_size!(0, DMA_BUFFER_SIZE, DMA_CHUNK_SIZE);
         let buffer = dma_alloc_buffer!(DMA_BUFFER_SIZE, DMA_ALIGNMENT as usize);
-        let mut dma_tx_buf =
-            DmaTxBuf::new_with_block_size(descriptors, buffer, Some(DMA_ALIGNMENT)).unwrap();
+        let mut dma_tx_buf = DmaTxBuf::new_with_config(
+            descriptors,
+            buffer,
+            BurstConfig {
+                internal: InternalBurstConfig::default(),
```

Comment:
maybe it's just me but `internal` and `external` are not very descriptive

Reply:
I'm wondering if this is better as an `enum` anyway. A buffer can't live both in external and internal memory; although a descriptor chain may contain both at the same time. But in that case, the descriptors themselves have to be configured according to that, and we can do that with an `enum` and a way to chain buffers.